novawallet/Modules/DApp/DAppList/DAppListPresenter.swift

```diff
@@ -16,16 +16,20 @@ final class DAppListPresenter {
     private var hasFavorites: Bool { !(favorites ?? [:]).isEmpty }
     private var hasWalletsListUpdates: Bool = false
 
+    private var dAppOpenTask: (() -> Void)?
+
     private lazy var iconGenerator = NovaIconGenerator()
 
     init(
         interactor: DAppListInteractorInputProtocol,
         wireframe: DAppListWireframeProtocol,
+        wallet: MetaAccountModel,
```

Comment:
It should come from the interactor I believe as a user might change wallet on that screen